### PR TITLE
Add error message when interpretation fails

### LINKED ISSUE #759: Improve discoverability of errors that break the alerts view
**Describe the bug**
In [a recent discussion](https://github.com/github/securitylab/discussions/274#discussioncomment-388467), @adriaanjacobs asked about a situation where the alerts view was silently unusable for a particular path query against a particular database. Ultimately, there was an error in the extension output pane, which indicated that we had [encountered an exception](https://github.com/github/vscode-codeql/blob/ab31d86a8d7e09926772e0b84744e8a45639be47/extensions/ql-vscode/src/interface.ts#L630) while processing interpreted results, and were falling back on raw display.

While that's certainly better than not displaying results at all, it would be nice to find a way of making it clear why the `alerts` view is non-functional (perhaps by replacing it with an error pane that shows details about the exception we caught?).

**To Reproduce**
I have a database and query that reproduce the problem -- ping me on Slack if it would be useful to have them.

### PR DESCRIPTION
**Pull request by aeisenberg**
One way it can fail is if the SARIF is too large. We explicitly call
out that error because the raw message received from the node runtime
is not very understandable.

Fixes #759

## Checklist

- [x] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [x] Issues have been created for any UI or other user-facing changes made by this pull request.
- [n/a] `@github/docs-content-dsp` has been cc'd in all issues for UI or other user-facing changes made by this pull request.

### CODE REVIEW

File: extensions/ql-vscode/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## [UNRELEASED]
 
 - Allow using raw LGTM project slugs for fetching LGTM databases. [#769](https://github.com/github/vscode-codeql/pull/769)
+- Better error messages when BQRS interpretation fails to produce SARIF. [#770](https://github.com/github/vscode-codeql/pull/770)
```
Comment (adityasharad) [inline extensions/ql-vscode/CHANGELOG.md:6]:
```suggestion
- Better error messages when result interpretation fails to produce SARIF for the `alerts` view . [#770](https://github.com/github/vscode-codeql/pull/770)
```

File: extensions/ql-vscode/src/cli.ts [hunk 2]
```diff
@@ -603,8 +603,12 @@ export class CodeQLCliServer implements Disposable {
     let output: string;
     try {
       output = await fs.readFile(interpretedResultsPath, 'utf8');
-    } catch (err) {
-      throw new Error(`Reading output of interpretation failed: ${err.stderr || err}`);
+    } catch (e) {
+      const rawMessage = e.stderr || e.message;
+      const errorMessage = rawMessage.startsWith('Cannot create a string')
+        ? `SARIF too large. ${rawMessage}`
```
Comment (adityasharad) [inline extensions/ql-vscode/src/cli.ts:609]: Could we log the raw message but make the user-facing one a bit friendlier? Something like "SARIF results are too large to process for the alerts view. Will show raw results instead."

**Review by adityasharad — APPROVED** [review]
Generally looks good. Minor suggestion on whether we can make the user-facing message shorter and more understandable, so they understand the exact impact.

File: extensions/ql-vscode/src/cli.ts (hunk 2 above)
Comment (aeisenberg) [reply extensions/ql-vscode/src/cli.ts:609]: Hmmm...I think I can do that.

**Pull request merged**